rclpy/rclpy/type_support.py

```diff
@@ -26,6 +26,34 @@ def check_for_type_support(msg_type):
             *e.args[1:])
         raise
     if ts is None:
-        msg_type.__class__.__import_type_support__()
-    if msg_type.__class__._TYPE_SUPPORT is None:
+        msg_or_srv_type.__class__.__import_type_support__()
+    if msg_or_srv_type.__class__._TYPE_SUPPORT is None:
         raise NoTypeSupportImportedException()
+
+
+def check_is_valid_msg_type(msg_type):
+    check_for_type_support(msg_type)
+    try:
+        assert None not in (
+            msg_type.__class__._CREATE_ROS_MESSAGE,
+            msg_type.__class__._CONVERT_FROM_PY,
+            msg_type.__class__._CONVERT_TO_PY,
+            msg_type.__class__._DESTROY_ROS_MESSAGE,
+        )
+    except (AssertionError, AttributeError):
+        raise RuntimeError(
+            f'The message type provided is not valid ({msg_type}),'
+            ' this might be a service or action'
+        ) from None
+
+
+def check_is_valid_srv_type(srv_type):
+    check_for_type_support(srv_type)
+    try:
+        assert None not in (
+            srv_type.Response,
+            srv_type.Request,
+        )
+    except (AssertionError, AttributeError):
+        raise RuntimeError(
+            f'The service type provided ({srv_type}), this might be a message or action') from None
```

Comment:
@ivanpauno nit: 
```suggestion
            f'The service type provided is not valid ({srv_type}),'
            ' this might be a message or action'
        ) from None
```